### PR TITLE
Fix test results summary

### PR DESCRIPTION
**Pull request by drakulavich**
#### Check List:
* Updates `EnricoMi/publish-unit-test-result-action` to v2;
* Fixes files pattern to publish JUnit test results
* Updates workflows to prevent double execution from PR (as `push` and `pull-request`)
* PR meets the [contributing guidelines](https://github.com/assertj/assertj/blob/main/CONTRIBUTING.md)

### CODE REVIEW

**Comment by drakulavich** [comment]
image

Fix for that summary ↑

**Comment by joel-costigliola** [comment]
That one was bothering but I did have the time to look into, thanks @drakulavich!

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,6 +1,9 @@
 name: CI
 
-on: [push, pull_request]
+on:
+  push:
+    branches: [ main ]
```
Comment (scordio) [inline .github/workflows/main.yml:5]: This way the workflow doesn't run for branches unless there is a pull request for it. Is there a reason why you disabled that?

Comment (joel-costigliola) [reply .github/workflows/main.yml:5]: merged too quickly I guess :sweat_smile:

Comment (scordio) [reply .github/workflows/main.yml:5]: 🚨👮‍♂️🚨😄 

It's anyway a small adjustment, I'll take care of it.

Comment (drakulavich) [reply .github/workflows/main.yml:5]: Sorry to break the flow for local branches. I intended to skip multiple executions when you start working inside the fork's PR. Before the change each commit triggers the workflows two times.

If you need the experiments with local branches I suggest adding some additional pattern like:
```yml
on:
  push:
    branches:
      - main
      - local-pattern-here-*
```

Comment (drakulavich) [reply .github/workflows/main.yml:5]:
```

  ┌────────────────────────────┐       ┌─────────────────────────────┐
  │   Push feature branch      ├──────►│   Create Pull Request       │
  └─────────────┬──────────────┘       └──────────────┬──────────────┘
                │                                     │
                │                                     │
                │                                     │
 ┌──────────────▼──────────────────┐  ┌───────────────▼────────────┐
 │   Push event triggers workflow  │  │ PR event triggers workflow │
 └─────────────────────────────────┘  └────────────────────────────┘

```
I mean just pipeline execution, the tests summary step will be executed in each pipeline separately.

Comment (scordio) [reply .github/workflows/main.yml:5]: This is true only for branches of this repo but not for forks.

Check [this PR](https://github.com/assertj/assertj/pull/2794/checks) for example, `CI` and `Cross-Version` are running only for `pull-request` events, leading to a total of 12 checks.

However, if you check a [dependabot PR](https://github.com/assertj/assertj/pull/2787/checks), `CI` and `Cross-Version` are running for both `push` and `pull-request` events, leading to a total of 22 checks.

Comment (scordio) [reply .github/workflows/main.yml:5]: Let's properly address this topic in #2810.

**Pull request merged**